maven-core/src/main/java/org/apache/maven/graph/DefaultGraphBuilder.java

```diff
@@ -401,18 +403,49 @@ else if ( selectedProject.isDirectory() )
             return projects;
         }
 
-        List<File> files = Arrays.asList( request.getPom().getAbsoluteFile() );
+        List<File> files = Collections.singletonList( request.getPom().getAbsoluteFile() );
         collectProjects( projects, files, request );
         return projects;
     }
 
-    private void collectProjects( List<MavenProject> projects, List<File> files, MavenExecutionRequest request )
-        throws ProjectBuildingException
+    private void collectProjects( List<MavenProject> projects, final List<File> files,
+                                  final MavenExecutionRequest request ) throws ProjectBuildingException
     {
-        ProjectBuildingRequest projectBuildingRequest = request.getProjectBuildingRequest();
+        final ProjectBuildingRequest projectBuildingRequest = request.getProjectBuildingRequest();
+
+        List<ProjectBuildingResult> results;
+        long start = System.currentTimeMillis();
+        if ( request.getDegreeOfConcurrency() <= 1 || Boolean.getBoolean( "maven.concurrentGraph.disable" ) )
+        {
+            results = projectBuilder.build( files, request.isRecursive(), projectBuildingRequest );
+        }
+        else
+        {
+            ForkJoinPool forkJoinPool = new ForkJoinPool( request.getDegreeOfConcurrency(),
```

Comment:
Can you please add a comment about the 'true' value, like 'true' /* why true... */

Reply:
removed. in this implementation FJP async mode is not critical, so i removed that flag for simplicity. initially idea was to enforce fifo for for incoming tasks, but because all tasks are joined, this is not critical here.